runhouse/cli_utils.py

```diff
@@ -0,0 +1,526 @@
+
+    server_util_info = (
+        f"CPU Utilization: {round(cluster_cpu_utilization, 2)}% | GPU Utilization: {round(cluster_gpu_utilization, 2)}%"
+        if has_cuda
+        else f"CPU Utilization: {round(cluster_cpu_utilization, 2)}%"
+    )
+    console.print(server_util_info)
+
+    # print the environments in the cluster, and the resources associated with each environment.
+    print_envs_info(env_servlet_processes, current_cluster)
+
+
+def print_bring_cluster_up_msg(
+    cluster_name: str, msg_prefix="Can't execute the command"
+):
+    from runhouse.main import console
+
+    console.print(
+        f"{msg_prefix} because [reset]{cluster_name} is not up. To bring it up, run [bold italic]`runhouse cluster up {cluster_name}`[/bold italic]."
+    )
+
+
+def get_cluster_or_local(cluster_name: str):
+    from runhouse.main import console
+
+    if cluster_name:
+        current_cluster = rh.cluster(name=cluster_name)
+        if not current_cluster.is_up():
+            console.print(
+                f"Cluster [reset]{cluster_name} is not up. If it's an on-demand cluster, you can run "
+                f"[reset][bold italic]`runhouse cluster up {cluster_name}`[/bold italic] to bring it up automatically."
+            )
+            raise typer.Exit(1)
+        try:
+            if current_cluster._http_client:
+                current_cluster._http_client.check_server()
+        except requests.exceptions.ConnectionError:
+            console.print(
+                f"Could not connect to the server on cluster {cluster_name}. Check that the server is up with "
+                f"[reset][bold italic]`runhouse cluster status {cluster_name}`[/bold italic] or [bold italic]`sky status -r`[/bold italic] for on-demand clusters."
+            )
+            raise typer.Exit(1)
+        return current_cluster
+
+    cluster_or_local = rh.here
+    if cluster_or_local == "file" and not cluster_name:
+        # If running outside the cluster must specify a cluster name
+        console.print("Missing argument `cluster_name`.")
+        return
+    elif not cluster_or_local:
+        console.print(
+            "\N{smiling face with horns} Runhouse Daemon is not running... \N{No Entry} \N{Runner}. "
+            "Start it with [reset][bold italic]`runhouse restart`[/bold italic] or specify a remote "
+            "cluster to poll with [reset][bold italic]`runhouse cluster status <cluster_name>`[/bold italic]."
+        )
+        raise typer.Exit(1)
+
+    else:
+        # we are inside the cluster
+        current_cluster = cluster_or_local  # cluster_or_local = rh.here
+
+    return current_cluster
+
+
+####################################################################################################
+# Cluster logs utils
+####################################################################################################
+
+
+class LogsSince(int, Enum):
+    one_min = 1
+    five_min = 5
+    ten_min = 10
+    one_hour = 60
```

Comment:
might be confusing to have a mix minutes / hours / days here, maybe just keep everything in munutes?